### ISSUE TITLE
Customize AgArch to receive messages via a ros topic

### ISSUE THREAD

**Issue by Rezenders** [jason-lang/jason_ros#28]
Override checkMail

**Comment by Rezenders** [comment]
Done at 9839514

**Comment by Rezenders** [comment]
.send(droneB, askOne, online(X), A, 2000);

fails at receiver

```
[Logs]    [7/30/2019, 2:54:58 PM] [jason] [Atom] The functor of an atom functor should not be null!
[Logs]    [7/30/2019, 2:54:58 PM] [jason] java.lang.Exception
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSyntax.Atom.<init>(Atom.java:29)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSyntax.Atom.<init>(Atom.java:24)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSemantics.TransitionSystem.applyProcMsg(TransitionSystem.java:367)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSemantics.TransitionSystem.applySemanticRuleSense(TransitionSystem.java:189)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSemantics.TransitionSystem.sense(TransitionSystem.java:1497)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.infra.centralised.CentralisedAgArch.sense(CentralisedAgArch.java:178)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.infra.centralised.CentralisedAgArch.reasoningCycle(CentralisedAgArch.java:213)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.infra.centralised.CentralisedAgArch.run(CentralisedAgArch.java:234)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at java.base/java.lang.Thread.run(Thread.java:834)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 
[Logs]    [7/30/2019, 2:54:58 PM] [jason] [droneB] *** ERROR in the transition system (deliberate). Circumstance:
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   E =[]
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   I =[]
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   A =null
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   MB=[]
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   RP=null
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   AP=null
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   SE=+!kqml_received(droneA,askOne,online(X),null)
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   SO=null
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   AI=null
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSemantics.Unifier.unifyTerms(Unifier.java:227)
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   PA={}
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   AE=null
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   SI=null
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSemantics.Unifier.unifiesNoUndo(Unifier.java:168)
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   FA=[].
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSemantics.Unifier.unifiesNoUndo(Unifier.java:90)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSemantics.Unifier.unifyTerms(Unifier.java:312)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSemantics.Unifier.unifiesNoUndo(Unifier.java:168)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] Creating a new C!
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSyntax.Plan.isRelevant(Plan.java:223)
[Logs]    [7/30/2019, 2:54:58 PM] [jason]   PI={}
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSemantics.Unifier.bind(Unifier.java:379)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] java.lang.NullPointerException
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSemantics.TransitionSystem.applyFindOp(TransitionSystem.java:513)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSemantics.TransitionSystem.applySemanticRuleDeliberate(TransitionSystem.java:211)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.asSemantics.TransitionSystem.deliberate(TransitionSystem.java:1519)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.infra.centralised.CentralisedAgArch.deliberate(CentralisedAgArch.java:187)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.infra.centralised.CentralisedAgArch.reasoningCycle(CentralisedAgArch.java:214)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at jason.infra.centralised.CentralisedAgArch.run(CentralisedAgArch.java:234)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 	at java.base/java.lang.Thread.run(Thread.java:834)
[Logs]    [7/30/2019, 2:54:58 PM] [jason] 
[Logs]    [7/30/2019, 2:55:00 PM] [jason] 	at jason.asSyntax.Atom.<init>(Atom.java:29)
```

**Comment by Rezenders** [comment]
Fixed at 5c3a7fe
